### PR TITLE
Allow ActiveRecord::QueryMethods#reselect to accept a hash

### PR DESCRIPTION
**Pull request by sampatbadhe**
### Motivation / Background

Recently Allows ActiveRecord::QueryMethods#select to accept a hash in [PR](https://github.com/rails/rails/pull/45612) , This PR will allows to accepts hash for ActiveRecord::QueryMethods#reselect similar to ActiveRecord::QueryMethods#select.

### Detail

Add ability to use hash with columns and aliases inside #reselect method similar to #select which introduced in https://github.com/rails/rails/pull/45612 by @alextrueman 

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Changes that are unrelated should be opened in separate PRs.
* [x] There are no typos in commit messages and comments.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [x] Feature branch is up-to-date with `main` (if not - rebase it).
* [x] Pull request only contains one commit for bug fixes and small features. If it's a larger feature, multiple commits are permitted but must be descriptive.
* [x] Tests are added if you fix a bug or add a feature.
* [x] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.
* [x] PR is not in a draft state.
* [ ] CI is passing.

### CODE REVIEW

File: activerecord/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,6 @@
+*   Allow ActiveRecord::QueryMethods#reselect to receive hash values, similar to ActiveRecord::QueryMethods#select
+
+    *Sampat Badhe*
```
Comment (byroot) [inline activerecord/CHANGELOG.md:3]: You are missing a space here.

**Review by byroot — APPROVED** [review]
LGTM, the CHANGELOG just need to be fixed. (make sure it's all in one commit).

Let me know when it's done and I'll merge.

File: activerecord/CHANGELOG.md (hunk 1 above)
Comment (byroot) [reply activerecord/CHANGELOG.md:3]: space -> blank line, sorry.

Comment (sampatbadhe) [reply activerecord/CHANGELOG.md:3]: Thank you for the review @byroot I made the requested change.

**Pull request merged**